src/node.cc

```diff
@@ -734,8 +755,19 @@ void StartExecution(Environment* env) {
   env->set_start_execution_function(Local<Function>());
 
   if (start_execution.IsEmpty()) return;
+
+  Local<Value> main_script_v;
+  if (main_script_id == nullptr) {
+    // TODO(joyeecheung): make this mandatory - we may also create an overload
+    // for main_script that is a Local<Function>.
```

Comment:
Just to make sure I understand this correctly, overload means an overload of `StartExecution()` that takes a `Local<Function>`? I like that idea. :)

Reply:
@addaleax Yeah, and maybe we could also use an overload with a C++ callback that takes some useful arguments (maybe Environment?)